### PR TITLE
Add controllers

### PR DESCRIPTION
**Pull request by Leeoasis**
### Completed Tasks in this PR

- [ ] Create controller for all users .
- [ ] Add route for all users.
- [ ] Create view for all users.
- [ ] Create controller single user .
- [ ]  Add route for single user.
- [ ] Create view for single user.
- [ ] Create controller for all posts .
- [ ] Add route for all posts.
- [ ] Create view for all posts.
- [ ] Create controller for a  single post.
- [ ] Add route for a single post.
- [ ]  Create view for a single post.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get 'posts/index'
+  get 'posts/show'
+  get 'users/index'
+  get 'users/show'
```
Comment (Gambit142) [inline config/routes.rb]: - [ ] Your application routes should be similar to that of the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md) provided. The routes in the wireframe look like these:
- http://localhost:3000/users/
- http://localhost:3000/users/745
- http://localhost:3000/users/745/posts
- http://localhost:3000/users/posts/3

#### Hint: To be able to create routes in this fashion, I will advise you to use the Rails resources as specified in this [article](https://guides.rubyonrails.org/routing.html#:~:text=2.7%20Nested%20Resources) to create your routes. Please, only create the needed routes. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Leeoasis 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Created controllers for User and Post model ✔️ 
- Created corresponding view files for controllers created ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
-  get 'posts/index'
-  get 'posts/show'
-  get 'users/index'
-  get 'users/show'
+  Rails.application.routes.draw do
+    resources :users, only: %i[index show] do
+      resources :posts, only: [:index]
+    end
+
+    get '/users/:user_id/posts/:id', to: 'posts#show'
```
Comment (Reem-lab) [inline config/routes.rb]: - [ OPTIONAL ] I think you can add it beside :index, :show like the users instead of adding it like this 🤩🎉💪🏻

**Review by Reem-lab — APPROVED** [review]
# Hi @Leeoasis,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the funtionality 💯
- [ ] Followed Gitflow 🔥
- [ ] No Linters error ✅
- [ ] Kudos for set up the controller⭐


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**